packages/flutter/lib/src/widgets/raster_widget.dart

```diff
@@ -0,0 +1,258 @@
+    required double devicePixelRatio,
+    required ValueNotifier<bool> rasterize,
+  }) : _delegate = delegate,
+       _devicePixelRatio = devicePixelRatio,
+       _rasterize = rasterize {
+    _currentlyRepaintBoundary = rasterize.value;
+  }
+
+  /// The device pixel ratio used to create the child image.
+  double get devicePixelRatio => _devicePixelRatio;
+  double _devicePixelRatio;
+  set devicePixelRatio(double value) {
+    if (value == devicePixelRatio) {
+      return;
+    }
+    _devicePixelRatio = value;
+    markNeedsPaint();
+  }
+
+  /// Whether a rasterized version of this render objects child is drawn in
+  /// place of the child.
+  ValueNotifier<bool> get rasterize => _rasterize;
+  ValueNotifier<bool> _rasterize;
+  set rasterize(ValueNotifier<bool> value) {
+    if (value == rasterize) {
+      return;
+    }
+    rasterize.removeListener(_onRasterValueChanged);
+    final bool oldValue = rasterize.value;
+    _rasterize = value;
+    rasterize.addListener(_onRasterValueChanged);
+    if (oldValue != rasterize.value) {
+      _onRasterValueChanged();
+    }
+  }
+
+  /// The delegate used to draw the image representing the child.
+  RasterWidgetDelegate get delegate => _delegate;
+  RasterWidgetDelegate _delegate;
+  set delegate(RasterWidgetDelegate value) {
+    if (value == delegate) {
+      return;
+    }
+    delegate.removeListener(markNeedsPaint);
+    final RasterWidgetDelegate oldDelegate = _delegate;
+    _delegate = value;
+    delegate.addListener(markNeedsPaint);
+    if (delegate.shouldRepaint(oldDelegate)) {
+      markNeedsPaint();
+    }
+  }
+
+  ui.Image? _childRaster;
+  bool _currentlyRepaintBoundary = false;
+
+  @override
+  void attach(covariant PipelineOwner owner) {
+    delegate.addListener(markNeedsPaint);
+    rasterize.addListener(_onRasterValueChanged);
+    super.attach(owner);
+  }
+
+  @override
+  void detach() {
+    delegate.removeListener(markNeedsPaint);
+    rasterize.removeListener(_onRasterValueChanged);
+    _childRaster?.dispose();
+    _childRaster = null;
+    super.detach();
+  }
+
+  @override
+  void dispose() {
+    delegate.removeListener(markNeedsPaint);
+    rasterize.removeListener(_onRasterValueChanged);
+    _childRaster?.dispose();
+    _childRaster = null;
+    super.dispose();
+  }
+
+  void _onRasterValueChanged() {
+    final bool wasRepaintBoundary = _currentlyRepaintBoundary;
+    _currentlyRepaintBoundary = rasterize.value;
+    if (_currentlyRepaintBoundary != wasRepaintBoundary) {
+      markNeedsCompositingBitsUpdate();
+    }
+    markNeedsPaint();
+  }
+
+  // Paint [child] with this painting context, then convert to a raster and detach all
+  // children from this layer.
+  ui.Image _paintAndDetachToGpuImage() {
+    final OffsetLayer offsetLayer = layer! as OffsetLayer;
+    final PaintingContext context = PaintingContext(offsetLayer, Offset.zero & size);
+    super.paint(context, Offset.zero);
+    // ignore: invalid_use_of_protected_member
```

Comment:
Ideally, we avoid this ignore altogether. At the very least it needs a comment explaining why this is here and whether this is temporary.

Reply:
Added comment, LMK what you think